patches/minecraft/net/minecraft/client/particle/ParticleManager.java.patch

```diff
@@ -49,7 +66,15 @@
           Iterable<Particle> iterable = this.field_78876_b.get(iparticlerendertype);
           if (iterable != null) {
              GlStateManager.color4f(1.0F, 1.0F, 1.0F, 1.0F);
-@@ -342,7 +344,7 @@
+@@ -316,6 +319,7 @@
+ 
+             for(Particle particle : iterable) {
+                try {
++                  if (!icamera.func_78546_a(particle.func_187116_l())) continue; // Forge: Don't render particles outside frustum.
```

Comment:
Can we be sure a particle is only rendered in it's bounding box? Maybe it would be good to add something like ignoreFrustemCheck in particle so modded particles can override it?